app/lib/search/index_simple.dart

```diff
@@ -282,7 +285,13 @@ class TokenIndex {
     }
     for (String id in counts.keys.toList()) {
       final double current = counts[id];
-      counts[id] = 100.0 * (current / _weights[id]) * (current / sumWeight);
+      double score = 100.0 * (current / sumWeight);
+      if (idfWeight > 0.0) {
+        // adjust score with inverse document frequency
+        final double idf = current / _weights[id];
```

Comment:
The inverse document frequency `idf` should tell us how special the term is among our corpus (for very unique terms it will be high, for common terms it will be very low).

For example in english text, the term `the` will be common among many documents so it will get a very low `idf` score. Whereas the term `geospatial` will have a very high `idf` because only very few documents contain that term. (Normally the logarithm of the ratio is taken, see [wikipedia](https://en.wikipedia.org/wiki/Tf%E2%80%93idf) )

This computation can be done when building the index not when querying it! After building the idf's for all terms when indexing, we can normalize them to be within `[0.0, 1.0]`.